google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProviderTest.java

```diff
@@ -644,6 +646,31 @@ public void testProviderEquals() {
     assertThat(path1.getFileSystem().provider()).isNotEqualTo(path3.getFileSystem().provider());
   }
 
+  @Test
+  public void testFromSpace() throws UnsupportedEncodingException {
+    // User should be able to create paths to files whose name contains a space.
+    // Traditional way 1: manually escape the spaces
+    Path path1 = Paths.get(URI.create("gs://bucket/with/a%20space"));
+    CloudStorageFileSystemProvider provider =
+        (CloudStorageFileSystemProvider)path1.getFileSystem().provider();
+    // Traditional way 2: use a library to escape the spaces
+    String escaped = UrlEscapers.urlFragmentEscaper().escape("gs://bucket/with/a space");
+    Path path2 = Paths.get(URI.create(escaped));
+    // Non-traditional way: use our convenience method to work around URIs not being allowed to
+    // contain spaces.
+    Path path3 = provider.getPath("gs://bucket/with/a space");
+    // All 3 should be equivalent
+    assertThat(path1.getFileSystem().provider()).isEqualTo(path2.getFileSystem().provider());
+    assertThat(path2.getFileSystem().provider()).isEqualTo(path3.getFileSystem().provider());
+    assertThat(path1.toUri()).isEqualTo(path2.toUri());
+    assertThat(path2.toUri()).isEqualTo(path3.toUri());
+
+    Path path4 = provider.getPath("gs://bucket/with/a%20percent");
+    String toString = path4.toString();
+    assertThat(toString).doesNotContain(" ");
+    assertThat(toString).contains("%");
```

Comment:
Changed now. The reason was that CloudStoragePath.toString() returns just the path (not the bucket name) and I didn't want to set that behavior in stone in this test.